clusterloader2/pkg/test/simpletestexecutor.go

```diff
@@ -27,21 +35,142 @@ func createSimpleTestExecutor() TestExecutor {
 }
 
 // ExecuteTest executes test based on provided configuration.
-func (ste *simpleTestExecutor) ExecuteTest(ctx Context, conf *api.Config) error {
-	return nil
+func (ste *simpleTestExecutor) ExecuteTest(ctx Context, conf *api.Config) []error {
+	defer cleanupResources(ctx)
+	var errList []error
+	err := ctx.GetFramework().CreateAutomanagedNamespaces(int(conf.AutomanagedNamespaces))
+	if err != nil {
+		errList = append(errList, fmt.Errorf("automanaged namespaces creation failed: %v", err))
+		return errList
+	}
+
+	for i := range conf.Steps {
+		if stepErrList := ste.ExecuteStep(ctx, &conf.Steps[i]); len(stepErrList) > 0 {
+			errList = append(errList, stepErrList...)
+			if isErrsCritical(stepErrList) {
+				return errList
+			}
+		}
+	}
+	return errList
 }
 
 // ExecuteStep executes single test step based on provided step configuration.
-func (ste *simpleTestExecutor) ExecuteStep(ctx Context, step *api.Step) error {
-	return nil
+func (ste *simpleTestExecutor) ExecuteStep(ctx Context, step *api.Step) []error {
+	var errList []error
+	if len(step.Measurements) > 0 {
+		// TODO: handle measurments
+	} else {
+		var wg wait.Group
+		var lock sync.Mutex
+		for i := range step.Phases {
+			phase := &step.Phases[i]
+			wg.Start(func() {
+				if phaseErrList := ste.ExecutePhase(ctx, phase); len(phaseErrList) > 0 {
+					lock.Lock()
+					defer lock.Unlock()
+					errList = append(errList, phaseErrList...)
+				}
+			})
+		}
+		wg.Wait()
+	}
+	return errList
 }
 
 // ExecutePhase executes single test phase based on provided phase configuration.
-func (ste *simpleTestExecutor) ExecutePhase(ctx Context, phase *api.Phase) error {
+func (ste *simpleTestExecutor) ExecutePhase(ctx Context, phase *api.Phase) []error {
+	var errList []error
+	nsList := createNamespacesList(phase.NamespaceRange)
+	for _, nsName := range nsList {
+		for j := len(phase.ObjectBundle) - 1; j >= 0; j-- {
```

Comment:
Why are you traversing the objectbundle in reverse direction?

Reply:
I assumed that if objects are created in some specific order, then maybe they should be deleted in a reversed order. Are there problems like A can't be removed because it's used by B?